modules/local/samtools/count/main.nf

```diff
@@ -0,0 +1,46 @@
+
+process SAMTOOLS_COUNT {
+    // Given a bam file, count the number of reads that aligned.
+
+    tag { meta.id }
+    label 'process_high'
+
+    container 'nciccbr/ccbr_ubuntu_base_20.04:v6'
+
+    input:
+        tuple val(meta), path(bam), path(bai)
+
+    output:
+        tuple val(meta), env('COUNT'),          emit: count
+        path("versions.yml"),                   emit: versions
+
+    when:
+        task.ext.when == null || task.ext.when
+
+    script:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def filter_flag = meta.single_end ? '-F 4' : '-f 2'
```

Comment:
@kopardev can you double-check my work here?

- for single-end, use `-F 4` to exclude unmapped reads
- for paired-end, use `-f 2` to include reads mapped in proper pair

(used https://broadinstitute.github.io/picard/explain-flags.html to figure out flags)